File: Backend/Iron-Bite.API/Entities/IngredientMeal.cs
using Iron_Bite.API.Models;

namespace Iron_Bite.API.Entities;

public class IngredientMeal : BaseEntity
{
	public required Guid MealId { get; init; }
	public required Meal Meal { get; init; }
	public required Guid IngredientId { get; init; }
	public required Ingredient Ingredient { get; init; }
	public required Quantity Quantity { get; set; }
}

File: Frontend/iron-bite/src/app/models/ingredient-meal.ts
import {IMeal} from "./meal";
import {Ingredient} from "./ingredient";
import {Quantity} from "./quantity";
import {IBaseEntity} from "./base-entity";

export interface IngredientMeal {
  id: IBaseEntity;
  mealId: string;
  meal: IMeal;
  ingredientId: string;
  ingredient: Ingredient;
  quantity: Quantity;
}
